src/cmap/connection.ts

```diff
@@ -317,6 +320,23 @@ export class Connection extends EventEmitter {
 
     let clusterTime = this.clusterTime;
     let finalCmd = Object.assign({}, cmd);
+    const inTransaction = session && (session.inTransaction() || isTransactionCommand(finalCmd));
+
+    // if an API version was declared, add the apiVersion option to every command, except:
+    // a. only in the initial command of a transaction
+    // b. only in a Cursor's initiating command, not subsequent getMore commands
+    if (
+      this.serverApi &&
```

Comment:
What do you think about extracting this into a separate method? I tend to personally prefer doing that with complex checks as it makes this code block a bit easier to read.